test/test_fisk.rb

```diff
@@ -58,6 +58,51 @@ def test_lea_with_negative_offset
     assert_match(/r8, (?:qword ptr )?\[r9 \- 0xa\]/, lea.op_str.to_s)
   end
 
+  def test_lazy
+    buf = StringIO.new(''.b)
+    patch_location = nil
+    fisk.nop
+    fisk.lazy { |pos|
+      patch_location = pos
```

Comment:
I have a branch for comments that I'm working on that returns from `write_to` a metadata object that includes comments and could include other things like lazy instruction locations - could be an alternative way to do it compared to callbacks?

Reply:
Actually ya, that sounds much better than using callbacks.  As long as there's a way to differentiate "lazy" instructions (or whatever they should be called).  I guess we could just give them names, like labels?

Actually what do you think about the metadata including `put_label` locations?  Maybe that would be enough, and we wouldn't need a new method.

Reply:
`put_label` yeah - my comments are done with `comment` and then you get metadata that is a list of comments for addresses. I should share the branch...